### ISSUE TITLE
Upgrade needle dependency

### ISSUE THREAD

**Issue by blimmer** [charltoons/hipchatter#24]
the version of needle this uses is quite old and causes a problem with pre-node 4.
```
/Users/blimmer/code/oss/ember-cli-deploy-hipchat/node_modules/hipchatter/node_modules/needle/node_modules/qs/lib/index.js:5
const Stringify = require('./stringify');
^^^^^
Use of const in strict mode.
/Users/blimmer/code/oss/ember-cli-deploy-hipchat/node_modules/hipchatter/node_modules/needle/node_modules/qs/lib/index.js:5
const Stringify = require('./stringify');
^^^^^
SyntaxError: Use of const in strict mode.
    at exports.runInThisContext (vm.js:73:16)
    at Module._compile (module.js:443:25)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
```

I worked around this by pegging qs to a lower version (`^5.2.0`), but this might cause other folks problems too.

**Comment by skylord123** [comment]
Having same issue after running `npm update` a couple hours ago.

**Comment by niconistal** [comment]
I'm having the same issue, is anyone working on this? How can I help?

**Comment by blimmer** [comment]
I haven't had time to work on this yet. The quick fix for your project is to peg `qs` to a 5.2.x version, but if you can't do that, we'd need to see if we can make the changes to upgrade to a newer version of needle on this project.
